processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -578,6 +579,43 @@ public Integer getDimensionIndex(String dimension)
     return dimensionOrder.get(dimension);
   }
 
+  public LinkedHashMap<String, Integer> getDimensionOrder()
+  {
+    return dimensionOrder;
```

Comment:
synchronize on dimensionOrder, return a copy? (dimensionOrder is not a thread safe map)